src/services/inlayHints.ts

```diff
@@ -64,8 +64,8 @@ namespace ts.InlayHints {
                 visitCallOrNewExpression(node);
             }
             else {
-                if (preferences.includeInlayFunctionParameterTypeHints && isFunctionExpressionLike(node)) {
-                    visitFunctionExpressionLikeForParameterType(node);
+                if (preferences.includeInlayFunctionParameterTypeHints && (isArrowFunction(node) || isFunctionExpression(node) || isSetAccessorDeclaration(node))) {

```

Comment:
Maybe an aside from this PR - it feels like anything that can be contextually typed should get this treatment. Basically whatever's in `hasContextSensitiveParameters`, right?

https://github1s.com/microsoft/TypeScript/blob/8f6ad87d9980d028b312924c9bca1ff92ddb02a8/src/compiler/checker.ts#L16395-L16399

Reply:
@DanielRosenwasser  I've added condition with `hasContextSensitiveParameters`